File: Frontend/lib/models/product.dart
class Product {
  final int? id;
  final String name;
  final double price;
  final String? category;
  final String? brand;
  final int? stock;
  final String? image;
  final Map<String, dynamic>? specs;

  // ✅ CORRECTO - todos los parámetros nombrados dentro de {}
  Product({
    this.id,
    required this.name,
    required this.price,
    this.category,
    this.brand,
    this.stock,
    this.image,
    this.specs,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json["id"],
      name: json["name"],
      price: (json["price"] as num).toDouble(),
      category: json["category"],
      brand: json["brand"],
      stock: json["stock"],
      image: json["image"],
      specs: json["specs"] is Map
          ? Map<String, dynamic>.from(json["specs"])
          : null, // ✅ cast seguro
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "name": name,
      "price": price,
      "category": category,
      "brand": brand,
      "stock": stock,
      "image": image,
      "specs": specs,
    };
  }
}

File: Backend/app/models/product.py
from sqlalchemy import JSON, Column, Integer, String, Float
from app.database import Base

class Product(Base):
    __tablename__ = "productos"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255))
    category = Column(String(100))
    brand = Column(String(100))
    price = Column(Float)
    stock = Column(Integer, default=0)
    image = Column(String(500))
    specs = Column(JSON)
    status = Column(String, default="active")